prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -183,21 +198,139 @@ class FHIRRouter(
+        val processingModeDefaultResult: Boolean by lazy {
+            evaluateFilterCondition(
+                processingModeFilterDefault, bundle,
+                false
+            )
+        }
+
+        fullElrReceivers.forEach { receiver ->
+            // get the receiver's organization, since we need to be able to find/combine the correct filters
+            val orgFilters = settings.findOrganization(receiver.organizationName)!!.filters
+
+            // Get the applicable filters, either receiver or organization level if there are no receiver filters
+
+            // NOTE: these could all be combined into a single `if` statement, but that would reduce readability and
+            // ability to debug this section so is being left split out like this
+
+            // JURIS FILTER
+            //  default: allowNone
+            var passes = evaluateFilterCondition(getJurisFilters(receiver, orgFilters), bundle, false)
+            // QUALITY FILTER
+            //  default: must have message id, patient last name, patient first name, dob, specimen type
+            //           must have at least one of patient street, zip code, phone number, email
+            //           must have at least one of order test date, specimen collection date/time, test result date
+            passes = passes &&
+                evaluateFilterCondition(getQualityFilters(receiver, orgFilters), bundle, qualFilterDefaultResult)
+            // ROUTING FILTER
+            //  default: allowAll
+            passes = passes &&
+                evaluateFilterCondition(getRoutingFilter(receiver, orgFilters), bundle, true)
+            // PROCESSING MODE FILTER
+            //  default: allowAll
+            passes = passes &&
+                evaluateFilterCondition(
+                    getProcessingModeFilter(receiver, orgFilters), bundle,
+                    processingModeDefaultResult
+                )
+
+            // if all filters pass, add this receiver to the list of valid receivers
+            if (passes) {
+                listOfReceivers.add(receiver)
+            }
+        }
+
+        return listOfReceivers
+    }
+
+    /**
+     * Takes a [bundle] and [filter], evaluates if the bundle passes the filter. If the filter is null,
+     * return [defaultResponse]
+     * @return Boolean indicating if the bundle passes the filter or not
+     */
+    internal fun evaluateFilterCondition(
+        filter: ReportStreamFilter?,
+        bundle: Bundle,
+        defaultResponse: Boolean
+    ): Boolean {
+        // the filter needs to check all expressions passed in, or if the filter is null or empty it will return the
+        // default response
+        return if (filter.isNullOrEmpty())
+            defaultResponse
+        else
+            filter.all {
+                FhirPathUtils.evaluateCondition(CustomContext(bundle, bundle), bundle, bundle, it)
```

Comment:
FYI: This will be where you pass in your constants (in the CustomContext) once you read them.